ESSArch_Core/WorkflowEngine/tests/test_tasks.py

```diff
@@ -236,3 +239,182 @@ def test_retry_with_args(self):
         res = task.retry().get()
 
         self.assertEqual(res, x + y)
+
+
+class DBTaskTests(TestCase):
+
+    @mock.patch("ESSArch_Core.WorkflowEngine.dbtask.get_cached_objid", return_value=mock.ANY)
+    @mock.patch("ESSArch_Core.WorkflowEngine.dbtask.logger.log")
+    def test_create_event_when_success(self, mocked_logger_log, mocked_get_cached_objid):
+        db_task = DBTask()
+        task_id = uuid.uuid4()
+
+        db_task.create_event(
+            task_id=task_id,
+            status=celery_states.SUCCESS,
+            args=["1", "2"],
+            kwargs={'some_key': 'some_value', '_options': 'opt_val'},
+            retval=None,
+            einfo=None
+        )
+
+        outcome_detail_note = ''
+        expected_extra = {
+            'event_type': db_task.event_type,
+            'object': db_task.ip,
+            'agent': None,
+            'task': task_id,
+            'outcome': EventIP.SUCCESS
+        }
+        mocked_logger_log.assert_called_once_with(logging.INFO, outcome_detail_note, extra=expected_extra)
+
+    @mock.patch("ESSArch_Core.WorkflowEngine.dbtask.get_cached_objid", return_value=mock.ANY)
+    @mock.patch("ESSArch_Core.WorkflowEngine.dbtask.logger.log")
+    @mock.patch("billiard.einfo.ExceptionInfo")
+    def test_create_event_when_failure(self, mock_einfo, mock_logger_log, mock_get_cached_objid):
+        db_task = DBTask()
+        task_id = uuid.uuid4()
+
+        db_task.create_event(
+            task_id=task_id,
+            status=celery_states.PENDING,
+            args=["1", "2"],
+            kwargs={'some_key': 'some_value', '_options': 'opt_val'},
+            retval=None,
+            einfo=mock_einfo
+        )
+
+        expected_extra = {
+            'event_type': db_task.event_type,
+            'object': db_task.ip,
+            'agent': None,
+            'task': task_id,
+            'outcome': EventIP.FAILURE
+        }
+        mock_logger_log.assert_called_once_with(logging.ERROR, mock.ANY, extra=expected_extra)
+
+    @mock.patch("ESSArch_Core.WorkflowEngine.models.ProcessTask.objects.filter")
+    @mock.patch("ESSArch_Core.WorkflowEngine.dbtask.DBTask.create_event")
+    def test_success_when_event_type_not_none_then_create_event(self, mock_create_event, mock_process_task):
+        db_task = DBTask()
+        db_task.event_type = not None
```

Comment:
`event_type` is either `None` or an `int`

```suggestion
        db_task.event_type = 123
```